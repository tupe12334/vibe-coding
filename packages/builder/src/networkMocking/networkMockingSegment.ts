import type { RootContent } from "mdast";

/**
 * Generate a line describing how network requests should be mocked.
 *
 * @param method - chosen mocking method
 * @returns markdown AST nodes representing the instruction
 */
export const networkMockingSegment = async (
  method: string
): Promise<RootContent[]> => {
  const segment: RootContent[] = [
    {
      type: "paragraph",
      children: [{ type: "text", value: `Network mocking: ${method}` }],
    },
  ];

  return segment;
};
