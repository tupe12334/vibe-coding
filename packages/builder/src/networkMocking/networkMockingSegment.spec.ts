import { describe, expect, test } from "vitest";
import { networkMockingSegment } from "./networkMockingSegment";

describe("networkMockingSegment", () => {
  test("returns network mocking line", async () => {
    const segment = await networkMockingSegment("msw");
    expect(segment).toMatchSnapshot();
  });
});
