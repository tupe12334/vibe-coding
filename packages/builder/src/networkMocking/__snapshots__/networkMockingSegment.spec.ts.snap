// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`networkMockingSegment > returns network mocking line 1`] = `
[
  {
    "children": [
      {
        "type": "text",
        "value": "Network mocking: msw",
      },
    ],
    "type": "paragraph",
  },
]
`;
